src/umpire/op/HostReallocateOperation.cpp

```diff
@@ -31,9 +33,15 @@ void HostReallocateOperation::transform(
 {
   auto allocator = dst_allocation->m_strategy;
 
-  ResourceManager::getInstance().deregisterAllocation(src_ptr);
-
   *dst_ptr = ::realloc(src_ptr, length);
+  
+  if (dst_ptr == src_ptr) {
```

Comment:
Was this a bug that was found while testing this?  Does it relate to using pool storage of an allocation record?

Reply:
It's not quite finished.